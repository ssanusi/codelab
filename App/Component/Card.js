import React from 'react';
import {
  View, Text, StyleSheet, Image, TouchableOpacity,
} from 'react-native';
import { secondary } from '../lib/constants';


const Card = ({
  fullName, avatarUrl, username, navigation,
}) => (
  <TouchableOpacity
    style={style.container}
    onPress={() => navigation.navigate('Profile', { username })}
  >
    <View style={style.flexLayout}>
      <Image style={style.thumbnail} source={{ uri: avatarUrl }} />
      <View>
        <Text>{fullName}</Text>
        <View style={style.text}>
          <Text>
            <Image style={style.logo} source={require('../../assets/GitHub-Mark-64px.png')} />
            {' '}
            {username}
          </Text>
        </View>
      </View>
      <View>
        <Image
          style={style.arrowIcon}
          source={require('../../assets/ca96d96471dd4fd694ab44f2d1246918.png')}
        />
      </View>
      <View />
    </View>
  </TouchableOpacity>
);

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    backgroundColor: secondary,
    borderRadius: 6.33,
    width: 331.41,
    height: 83.65,
    marginHorizontal: 18,
    marginTop: 15,
  },
  flexLayout: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  thumbnail: {
    height: 60,
    width: 60,
    borderRadius: 25,
  },
  arrowIcon: {
    width: 30,
    height: 30,
  },
  logo: {
    height: 12,
    width: 12,
  },
  text: {
    padding: 3,
  },
});

export default Card;
