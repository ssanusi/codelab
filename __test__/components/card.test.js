import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../App/Component/Card';

it('Card Component renders without crashing', () => {
  const CardRendered = renderer.create(<Card />).toJSON();
  expect(CardRendered).toBeTruthy();
  expect(CardRendered).toMatchSnapshot();
});
