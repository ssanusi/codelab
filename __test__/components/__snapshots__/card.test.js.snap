// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders without crashing 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 6.33,
      "flex": 1,
      "flexDirection": "row",
      "height": 83.65,
      "justifyContent": "space-evenly",
      "marginHorizontal": 18,
      "marginTop": 15,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 25,
          "height": 60,
          "width": 60,
        }
      }
    />
    <View>
      <Text />
      <View
        style={
          Object {
            "padding": 3,
          }
        }
      >
        <Text>
          <Image
            source={1}
            style={
              Object {
                "height": 12,
                "width": 12,
              }
            }
          />
           
        </Text>
      </View>
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
    <View />
  </View>
</View>
`;
